### PR TITLE
Handle combined expect_out and absent_out

### LINKED ISSUE #6735: Correctly handle expect_out and absent_out together in the RunApp tester
### Description of the enhancement or error report
There is currently a logic error in the RunApp tester. If a test asks for both `expect_err` and `absent_err` to be checked, it only checks the `expect_err` criterion.  If the text in the `expect_err` criterion is present, it will pass even if the text in `absent_err` is present.

### Rationale for the enhancement or information for reproducing the error
The test that I added in #6734 uses a combination of `expect_err` and `absent_err` to ensure that the nonlinear iterations are within certain limits. It passes in its current state, but if one were to force more nonlinear iterations by decreasing `max_increment`, it would still pass even though the text specified in `absent_err` would be there in that case.

### Identified impact
There are a few tests that use both of these checks that may be incorrectly reporting that they are passing currently.

### PR DESCRIPTION
**Pull request by bwspenc**
closes #6735

### CODE REVIEW

**Comment by bwspenc** [comment]
Note that this gets rid of the 'OUTPUT ABSENT BUT CRASH' and 'FOUND OUT BUT CRASH' cases. I think there would be too many of these, with no real purpose ('FOUND OUT AND OUTPUT ABSENT BUT CRASH' ???).  If it crashes, you just get the 'CRASH' reason now.

File: python/TestHarness/testers/RunApp.py [hunk 1]
```diff
@@ -235,18 +235,12 @@ def processResults(self, moose_dir, retcode, options, output):
...
 
-    elif specs.isValid('absent_out'):
-      out_ok = self.checkOutputForPattern(output, specs['absent_out'])
-      # Process out_ok
-      if (not out_ok and retcode != 0):
-        reason = 'OUTPUT ABSENT BUT CRASH'
-      elif (out_ok):
+    if reason == '' and specs.isValid('absent_out'):
+      out_ok = not self.checkOutputForPattern(output, specs['absent_out'])
+      if (not out_ok):
```
Comment (jwpeterson) [inline python/TestHarness/testers/RunApp.py]: The double-negative on lines 242 and 243 is kind of confusing, but I think it's correct...

**Comment by jwpeterson** [comment]
Unrelated: "NO EXPECTED OUT" has always confused me.  Does it mean, "there is no expected output, but we got some" or "expected output not found"?  It's the latter, so maybe we could change it slightly.  'EXPECTED OUTPUT MISSING'?

File: python/TestHarness/testers/RunApp.py (hunk 1 above)
Comment (bwspenc) [inline python/TestHarness/testers/RunApp.py]: I don't think it's not correct.

**Comment by bwspenc** [comment]
@jwpeterson That's a good suggestion.  I'll change it.

**Comment by bwspenc** [comment]
@jwpeterson I just made these revisions:
- Changed variable names to be more clear and got rid of double negative
- Report 'EXPECTED OUTPUT MISSING' for error

**Comment by jwpeterson** [comment]
Looks good.  We can wait a little while to see if @permcody disagrees before merging.

**Comment by permcody** [comment]
I'm fine reorganizing a bit. I suppose it's also fine to say a crash is a crash and basically invalidates the rest of the test unless it's supposed to crash (`mooseError()`). I know in the past some of our apps behaved pretty badly and while they ran a simulation mostly fine, they'd often crash during cleanup. I put some of these extra cases in to handle that (OUTPUT FOUND BUT CRASH).

**Comment by bwspenc** [comment]
@permcody Sure, I understand why it was giving more information (OUTPUT FOUND BUT CRASH).  The problem is that when we check for both finding the output and that the output is absent, there starts to be a lot of cases:
1. found output and output absent but crash
2. found output but output not absent but crash
3. didn't find output and output absent but crash
It just seemed a little silly to report all of these cases (and hard to come up with a concise string that makes sense to describe what happened), but if you think it's helpful to include them still, I can add that.  I'm fine either way.

**Comment by permcody** [comment]
👍

**Pull request merged**